File: Warehousing.Data/Entities/StoreTransferItem.cs
namespace Warehousing.Data.Entities
{
    public class StoreTransferItem : BaseClass
    {
        public int Id { get; set; }
        
        public int TransferId { get; set; }
        public StoreTransfer Transfer { get; set; } = null!;
        
        public int ProductId { get; set; }
        public Product Product { get; set; } = null!;
        
        public decimal Quantity { get; set; }
        public decimal UnitCost { get; set; }  // Track cost for accounting
        public string Notes { get; set; } = string.Empty;
    }
}

File: Warehousing.UI/src/app/admin/models/storeTransferItem.ts
import { StoreTransfer } from "./storeTransfer";
import { Product } from "./product";

export interface StoreTransferItem {
    id: number;
    transferId: number;
    transfer: StoreTransfer;
    productId: number;
    product: Product;
    quantity: number;
    unitCost: number;
    notes: string;
    createdAt: string | null;
    createdBy: string | null;
    updatedAt: string | null;
    updatedBy: string | null;
}

export interface StoreTransferItemDto {
    id: number | null;
    transferId: number | null;
    productId: number;
    quantity: number;
    unitCost: number;
    notes: string;
}



